pgjdbc/src/main/java/org/postgresql/jdbc/SQLDataReader.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2024, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.jdbc;
+
+import static java.lang.Character.isWhitespace;
+
+import org.postgresql.core.BaseConnection;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.sql.SQLData;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SQLDataReader {
+  public @Nullable  <T> T read(@Nullable String value, Class<T> type, BaseConnection connection, TimestampUtils timestampUtils) throws SQLException {
+    if (value == null) {
+      return null;
+    }
+    SQLData data;
+    try {
+      data = (SQLData) type.getConstructor().newInstance();
+    } catch (Exception ex) {
+      throw new SQLException(String.format("An accessible no-arg constructor is required for type [%s]", type), ex);
```

Comment:
Use PSQLException(GT.tr("An accessible no-arg constructor is required for type {0}", type" ... so we can have translations. State might be SYNTAX_ERROR

Reply:
Done